documentation_v12/Types/Enum_Type.md

```diff
@@ -46,65 +61,183 @@ If quality is not explicitly set then it is assumed to be \'good\'.
+
+    { 
+		"id": "ConfigurationFlags",
+		"flags": {
+			"values": [ 
+                		{ "name":"CONFIG_BIT01", "value":1 }, 
+                		{ "name":"CONFIG_BIT02", "value":2 },
+                		{ "name":"CONFIG_BIT03", "value":4 },
+                		{ "name":"CONFIG_BIT04", "value":8 },
+                		{ "name":"CONFIG_BIT05", "value":16 },
+				{ "name":"CONFIG_BIT06", "value":1024 }  
+			] 
+		}
+	}
 
+	{ 
+		"id": "ConfigurationFlags",
+		"flags": {
+			"values": [ 
+                		{ "value":1, "name":"CONFIG_BIT01" }, 
+                		{ "value":2, "name":"CONFIG_BIT02" },
+                		{ "value":4, "name":"CONFIG_BIT03" },
+                		{ "value":8, "name":"CONFIG_BIT04" },
+                		{ "value":16,"name":"CONFIG_BIT05" },
+				{ "value":1024, "name":"CONFIG_BIT06" }  
+ 			] 
+		}
+	}
+
+    	{ 
+		"id": "ConfigurationFlags",
+		"flags": {
+			"type": "Integer",
+			"format": "int32",
+			"values": [ 
+                		{ "name":"CONFIG_BIT01", "value":1 }, 
+                		{ "name":"CONFIG_BIT02", "value":2 },
+                		{ "name":"CONFIG_BIT03", "value":4 },
+                		{ "name":"CONFIG_BIT04", "value":8 },
+                		{ "name":"CONFIG_BIT05", "value":16 },
+				{ "name":"CONFIG_BIT06", "value":1024 }  
+ 			] 
+		}
+	}
+
+## Flags Type Messages for Data Quality
+
+The following Type message defines a data quality `flags` where multiple states (bits) can be active at the same time.
+
+A `flags` type could be used to define a set of allowed values that represents data quality. In this case include the `quality` in the `flags` values definition. 
+If quality is not explicitly set then it is assumed to be \'good\'. Flags can combine multiple states with different quality mappings set:
+ - When at least one flag with bad quality is set the value is considered bad.
+ - When at least one flag with questionable quality is set the value is considered questionable.
+ - Bad and questionable bits can be combined but bad does take precedence.
+
+```
+{ 
+	"id": "ConfigurationQualityFlags", 
+	"flags": {
+		"values": [ 
+            		{ "name":"Good", "value":1 }, 
+            		{ "name":"Uninitialized", "value":2, "quality": "bad" },
+            		{ "name":"Unreasonable", "value":4, "quality": "bad" },
+            		{ "name":"Suspect", "value":8, "quality": "questionable" },
+            		{ "name":"Commanded", "value":16 } 
+        	] 
+	}
+}
+```
+
+When referencing the `flags` enum from the property that holds quality, include the [isquality](xref:typePropertiesAndFormats) keyword and use `reftypeid` as the data type of the property:
+
+    { "DeviceStatus": { "reftypeid": "ConfigurationQualityFlags", "isquality": true } }
+
+## Remarks
+ - Properties referencing `flags` or `enum` type must send integer values in a data message, since use of strings is unsupported.
```

Comment:
Suggest: Properties referencing `flags` or `enum` type must send integer values in data messages; the use of strings is unsupported.

Reply:
Suggestion applied. Thank you David.